### DISCUSSION TITLE
Simplify result error

### DISCUSSION THREAD

**Discussion by Vernlium** [halide/Halide#6709] [Q&A] [answered]
Hi all,

I have a expr like this:

```c++
(2 - ((min(Output1_extent_0 - (Output1_s0_v0_v2 *17), 17) + 31)/32))
```

after simplify, expr change to:

```c++
((max((Output1_s0_v0_v2 *17) - Output1_extent_0, -17)/32) + 2)
```

Set `Output1_extent_0 = 24`, `Output1_s0_v0_v2 [min = 0; max value = (Output1_extent_0 + 17 - 1) / 17 - 1 = 2 - 1 = 1]`.

when `Output1_s0_v0_v2 = 0`:

expr_original = `2 - ((min(24 - 0 * 17, 17) + 31) / 32) = 2 - ((17 + 31) / 32) = 2 - 1 = 1`

expr_simplify = `(max(0 * 17 - 24, -17) / 32) + 2 = (-17) / 32 + 2 = 0 + 2 = 2`

when `Output1_s0_v0_v2 = 1`:

expr_original = `2 - ((min(24 - 1 * 17, 17) + 31) / 32) = 2 - ((7 + 31) / 32) = 2 - 1 = 1`

expr_simplify = `(max(1 * 17 - 24, -17) / 32) + 2 = (-7) / 32 + 2 = 0 + 2 = 2`

the values are not equal.

So, is there a bug of Halide simplify? Or, How to express the expr to get the right result?

**Comment by abadams** [comment, marked as answer]
Halide defines division to round towards negative infinity, so (-7)/32 = -1

**Reply by Vernlium** [reply]
Thank you for your reply. Is there some way than I can get the result what I want? as well as (-7)/32 = 0.

**Reply by abadams** [reply]
Yeah, if you want division rounding towards zero, you can say div_round_to_zero(x, y) instead of x/y
